### PR TITLE
Rename `material-3-codelab` to `animated-responsive-layout`

### PR DESCRIPTION
**Pull request by domesticmouse**
## Pre-launch Checklist

- [x] I read the [Effective Dart: Style] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Effective Dart: Style]: https://dart.dev/guides/language/effective-dart/style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: animated-responsive-layout/step_06/android/app/src/debug/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (a-wallen) [inline animated-responsive-layout/step_06/android/app/src/debug/AndroidManifest.xml:1]: This file looks completely new. Is there any reason why it was recently added?

File: animated-responsive-layout/step_07/android/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,71 @@
+def localProperties = new Properties()
```
Comment (a-wallen) [inline animated-responsive-layout/step_07/android/app/build.gradle:1]: This too.

Comment (a-wallen) [reply animated-responsive-layout/step_07/android/app/build.gradle:1]: and `animated-responsive-layout/step_07/android/app/src/debug/AndroidManifest.xml`, `animated-responsive-layout/step_07/android/app/src/main/kotlin/com/example/animated_responsive_layout/MainActivity.kt`

**Review by a-wallen — APPROVED** [review]
Just based off the diffs, it looks like there are some completely new files.

File: animated-responsive-layout/step_06/android/app/src/debug/AndroidManifest.xml (hunk 1 above)
Comment (domesticmouse) [reply animated-responsive-layout/step_06/android/app/src/debug/AndroidManifest.xml:1]: For some reason, the matching file under `material-3-codelab` is showing as a file deletion. I suspect the rename inference logic failed in the case of repeated identical files.

**Comment by domesticmouse** [comment]
> Just based off the diffs, it looks like there are some completely new files.

Keep scrolling down, you'll see right at the bottom that there are a bunch of straight deletions that hopefully match up. Given it is a 800 file patch I'm not going to confirm. For what it's worth most of these files are output from a [`codelab_rebuild`](https://github.com/flutter/codelabs/tree/main/tooling/codelab_rebuild) run, so I'm not overly concerned. I verified the deltas on the first hundred of so files, and they are sane.

**Pull request merged**